### PR TITLE
Sign state in the background and automatically upon initialization

### PR DESCRIPTION
**Pull request by Ejhfast**
Adds new initialization to client, so will be a breaking change

### CODE REVIEW

File: nash-native-client/src/ws_client/client.rs [hunk 1]
```diff
@@ -578,6 +575,124 @@ impl Client {
     }
 }
 
+
+// Adding another layer of abstraction on top of client make it easier/safer to manage concurrent usage of the client
+// In particular this is necessary to spawn a process in the background that signs state
+pub struct Client {
+    inner: Arc<Mutex<InnerClient>>,
+    sign_loop_status: Arc<Mutex<bool>>,
```
Comment (notdanilo) [inline nash-native-client/src/ws_client/client.rs:583]: Where are we reading this value?
Can it be made `Arc<AtomicBool>`?

**Review by notdanilo — COMMENTED** [review]
Looks good. I just have a minor question regarding the `sign_loop_status` variable.

File: nash-native-client/src/ws_client/client.rs (hunk 1 above)
Comment (Ejhfast) [reply nash-native-client/src/ws_client/client.rs:583]: yep i think so, good suggestion

Comment (Ejhfast) [reply nash-native-client/src/ws_client/client.rs:583]: i actually don't think the value can be mutated easily across threads, so leaving it be for now. can do a small PR later if I'm wrong about that (very possible)

**Pull request merged**